### PR TITLE
E8 ocp revisions

### PR DESCRIPTION
**Pull request by shaneboulden**
This PR improves the RHCOS/OCP profile alignment with the Australian Cyber Security Centre (ACSC) Essential Eight guidance and RHCOS controlled immutability/platform management:

- Add additional controls for SCC/RBAC configuration
- Remove controls for packages not supported with RHCOS
- Remove file system integrity controls (handled by the File Integrity Operator)
- Remove RHCOS software update controls (handled by the Cluster Version Operator)
- Remove rsyslog packages from RHCOS configuration (handled by the `ClusterLogForwarder` CR)
- Remove firewalld configuration from RHCOS nodes (not included)
- Use NOSHA1 crypto policy to better align with ASD crypto guidelines

### CODE REVIEW

**Review by JAORMX — COMMENTED** [review]
How complete and tested is the ocp4 e8 profile? I could add CI for this if it's in a good state.

**Comment by shaneboulden** [comment]
I think it's as complete as can be, given the current guidance. The only tests added so far are the e2e ocp4 tests: https://github.com/ComplianceAsCode/content/pull/6026

**Comment by JAORMX** [comment]
@shaneboulden I'll add an e2e test for the ocp4 e8 profile. The rhcos4 e8 one is good to go. I'll get to this today.

**Comment by JAORMX** [comment]
@shaneboulden @jhrozek @mrogers950 adding CI for the ocp4-e8 profile https://github.com/openshift/release/pull/15482

**Comment by JAORMX** [comment]
/test all

retriggering again cause there was an issue setting up the new ocp4-e8 test.

File: rhcos4/profiles/e8.profile [hunk 1]
```diff
@@ -82,6 +82,6 @@ selections:
   - sshd_enable_strictmodes
 
   # See also: https://www.cyber.gov.au/ism/guidelines-using-cryptography
-  - var_system_crypto_policy=default
+  - var_system_crypto_policy=default_nosha1
```
Comment (jhrozek) [inline rhcos4/profiles/e8.profile:85]: Only one question: Why was the crypto policy changed from future to default and then default minus sha1?
btw also the link about the rules no longer works

Comment (shaneboulden) [reply rhcos4/profiles/e8.profile:85]: Thanks @jhrozek for taking a look at this, I've updated the refs. 

The [ASD crypto guidelines](https://www.cyber.gov.au/acsc/view-all-content/guidance/asd-approved-cryptographic-algorithms) only allow for SHA-2. When we first created the e8 profile for RHEL, the only available policy that prevented SHA-1 being used was 'future'. https://github.com/ComplianceAsCode/content/pull/5024

The 'NO-SHA1' policy module was subsequently added, and the e8 profile was revised. https://github.com/ComplianceAsCode/content/commit/9aa23b0d0dd9632096d97d75e5e5feffb9954001

RHCOS followed a similar pattern - 'NO-SHA1' wasn't available initially, and 'future' was used instead. Now that 'default+NO-SHA1' is available, it's a better fit. 

'default' was an oversight - it should have been 'future' -> 'default:NO-SHA1'. I initially pushed this, and then rebased over the top.

**Comment by shaneboulden** [comment]
e2e tests look great, thanks @JAORMX!

**Comment by JAORMX** [comment]
lgtm, I'll wait for @jhrozek to merge.

File: rhcos4/profiles/e8.profile (hunk 1 above)
Comment (jhrozek) [reply rhcos4/profiles/e8.profile:85]: Thank you for the detailed reply and fixing the reference.

**Review by jhrozek — APPROVED** [review]
/lgtm

**Pull request merged**